app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -277,6 +287,10 @@ public void onClick(View v) {
         if (!isUsingActionBar()) {
             searchbox.setText(lastQueryString);
         }
+
+        if (!this.containsHereFunction) {
+            hereFunctionHandler.refreshLocation();
+        }

```

Comment:
Ahh, okay! It is a nonintuitive to be clearing the location in `onCreate` based on whether the entity loading has evaluated a `here()` xpath function yet or not. It seems like you want to clear the location every time you enter a newly launched `EntitySelectActivity`. This can be determined by checking if the `Bundle` argument passed into `onCreate` is `null`, meaning it was a new launch, not a screen rotation.